### PR TITLE
[bugfix] fixed gdal build vrt : argument list too long

### PR DESCRIPTION
**Pull request by dumasl**

### CODE REVIEW

File: s2p.py [hunk 1]
```diff
@@ -504,8 +504,17 @@ def global_dsm(tiles):
     out_dsm_vrt = os.path.join(cfg['out_dir'], 'dsm.vrt')
     out_dsm_tif = os.path.join(cfg['out_dir'], 'dsm.tif')
 
-    common.run("gdalbuildvrt -vrtnodata nan  %s %s" % (out_dsm_vrt,
-                                                       " ".join(dsm_list)))
+    cmd = ['gdalbuildvrt', '-vrtnodata', 'nan', out_dsm_vrt]
+    cmd += dsm_list
+
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, stdin=subprocess.PIPE)
```
Comment (carlodef) [inline s2p.py]: This doesn't fix the problem. Try to run

    import subprocess
    subprocess.Popen(['echo'] + ['aaa'] * 1234567)

in your Python interpreter. On my machine it raises `OSError: [Errno 22] Invalid argument` in python 2, and `OSError: [Errno 7] Argument list too long` in Python 3. This is the same behaviour I get with the `common.run` wrapper

**Review by carlodef — COMMENTED** [review]
Hi dumasl, thank you for spotting this issue. Unfortunately your fix doesn't solve the problem on my machine. The only fix I know would be to have `gdalbuildvrt` read its arguments from `stdin`, but I don't know if `gdal` supports this.

File: s2p.py (hunk 1 above)
Comment (dyoussef) [reply s2p.py]: We can try using "xargs"? https://github.com/MISS3D/s2p/pull/93/commits/8995fcf5f9e870825a9f5b7bd2f0cfc55777c4f6

**Comment by dyoussef** [comment]
Now gdalbuildvrt reads its arguments from stdin (using xargs) https://github.com/MISS3D/s2p/pull/93/commits/8995fcf5f9e870825a9f5b7bd2f0cfc55777c4f6

**Review by carlodef — APPROVED** [review]

**Review by dyoussef — APPROVED** [review]

**Pull request merged**